File: server/php/src/Application/Domain/Payments/PaymentDetails/PaymentReferences.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class PaymentReferences
{
    public ?string $merchantReference = null;
    public ?string $merchantParameters = null;

    public function toArray(): array
    {
        return [
            'merchantReference' => $this->merchantReference,
            'merchantParameters' => $this->merchantParameters,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/payment_references.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class PaymentReferences
          attr_accessor :merchant_reference, :merchant_parameters

          def initialize(merchant_reference: nil, merchant_parameters: nil)
            @merchant_reference = merchant_reference
            @merchant_parameters = merchant_parameters
          end

          def to_h
            {
              'merchantReference' => @merchant_reference,
              'merchantParameters' => @merchant_parameters
            }
          end
        end
      end
    end
  end
end